import error with protobuf==4.21.0

## 🐛 Bug

import error when using new [protobuf](https://pypi.org/project/protobuf/#history) version

### To Reproduce

pip install pytorch_lightning protobuf

python
```
import pytorch_lightning as pl
```

Results
```
import pytorch_lightning as pl
/usr/local/lib/python3.9/site-packages/pytorch_lightning/__init__.py:30: in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
/usr/local/lib/python3.9/site-packages/pytorch_lightning/callbacks/__init__.py:26: in <module>
    from pytorch_lightning.callbacks.pruning import ModelPruning
/usr/local/lib/python3.9/site-packages/pytorch_lightning/callbacks/pruning.py:[31](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:32): in <module>
    from pytorch_lightning.core.lightning import LightningModule
/usr/local/lib/python3.9/site-packages/pytorch_lightning/core/__init__.py:16: in <module>
    from pytorch_lightning.core.lightning import LightningModule
/usr/local/lib/python3.9/site-packages/pytorch_lightning/core/lightning.py:40: in <module>
    from pytorch_lightning.loggers import LightningLoggerBase, LoggerCollection
/usr/local/lib/python3.9/site-packages/pytorch_lightning/loggers/__init__.py:18: in <module>
    from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
/usr/local/lib/python3.9/site-packages/pytorch_lightning/loggers/tensorboard.py:26: in <module>
    from torch.utils.tensorboard import SummaryWriter
/usr/local/lib/python3.9/site-packages/torch/utils/tensorboard/__init__.py:10: in <module>
    from .writer import FileWriter, SummaryWriter  # noqa: F401
/usr/local/lib/python3.9/site-packages/torch/utils/tensorboard/writer.py:9: in <module>
    from tensorboard.compat.proto.event_pb2 import SessionLog
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/event_pb2.py:17: in <module>
    from tensorboard.compat.proto import summary_pb2 as tensorboard_dot_compat_dot_proto_dot_summary__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/summary_pb2.py:17: in <module>
    from tensorboard.compat.proto import tensor_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_pb2.py:16: in <module>
    from tensorboard.compat.proto import resource_handle_pb2 as tensorboard_dot_compat_dot_proto_dot_resource__handle__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/resource_handle_pb2.py:16: in <module>
    from tensorboard.compat.proto import tensor_shape_pb2 as tensorboard_dot_compat_dot_proto_dot_tensor__shape__pb2
/usr/local/lib/python3.9/site-packages/tensorboard/compat/proto/tensor_shape_pb2.py:[36](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:37): in <module>
    _descriptor.FieldDescriptor(
/usr/local/lib/python3.9/site-packages/google/protobuf/descriptor.py:[56](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true#step:4:57)0: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot not be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```
### Expected behavior

No import error

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): pytorch-lightning-1.6.3
- PyTorch Version (e.g., 1.10): torch-1.11.0
- Python version (e.g., 3.9): 3.9
- OS (e.g., Linux): from docker image `python:3.9-slim`
- CUDA/cuDNN version:
- GPU models and configuration: CPU
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information: I also get this error on my macbook


### Additional context

Filed on [this github action](https://github.com/openclimatefix/nowcasting_forecast/runs/6607770425?check_suite_focus=true)


cc @tchaton @rohitgr7

--- a/pytorch_lightning/setup_tools.py
+++ b/pytorch_lightning/setup_tools.py
@@ -32,9 +32,10 @@ def _load_requirements(
     reqs = []
     for ln in lines:
         # filer all comments
+        comment = ""
         if comment_char in ln:
+            comment = ln[ln.index(comment_char) :]
             ln = ln[: ln.index(comment_char)]
-        comment = ln[ln.index(comment_char) :] if comment_char in ln else ""
         req = ln.strip()
         # skip directly installed dependencies
         if not req or req.startswith("http") or "@http" in req:

--- a/requirements/base.txt
+++ b/requirements/base.txt
@@ -8,3 +8,4 @@ torchmetrics>=0.4.1, <=0.7.2
 pyDeprecate>=0.3.1, <=0.3.2
 packaging>=17.0, <=21.3
 typing-extensions>=4.0.0, <4.2.1
+protobuf<=3.20.1  # strict. TODO: Remove after tensorboard gets compatible https://github.com/tensorflow/tensorboard/issues/5708

--- a/requirements/strategies.txt
+++ b/requirements/strategies.txt
@@ -1,5 +1,5 @@
 fairscale>=0.4.5, <=0.4.6
-deepspeed<0.7.0
+deepspeed<0.6.0
 # no need to install with [pytorch] as pytorch is already installed
 horovod>=0.21.2,!=0.24.0, <=0.24.3
 hivemind>=1.0.1, <=1.0.1; sys_platform == 'linux'

--- a/tests/callbacks/test_quantization.py
+++ b/tests/callbacks/test_quantization.py
@@ -20,6 +20,7 @@ from torch.quantization import FakeQuantizeBase
 from torchmetrics.functional import mean_absolute_percentage_error as mape
 
 from pytorch_lightning import seed_everything, Trainer
+from pytorch_lightning.accelerators import GPUAccelerator
 from pytorch_lightning.callbacks import QuantizationAwareTraining
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from pytorch_lightning.utilities.memory import get_model_size_mb
@@ -35,9 +36,14 @@ from tests.helpers.simple_models import RegressionModel
 @RunIf(quantization=True)
 def test_quantization(tmpdir, observe: str, fuse: bool, convert: bool):
     """Parity test for quant model."""
+    cuda_available = GPUAccelerator.is_available()
+
+    if observe == "average" and not fuse and GPUAccelerator.is_available():
+        pytest.xfail("TODO: flakiness in GPU CI")
+
     seed_everything(42)
     dm = RegressDataModule()
-    accelerator = "gpu" if torch.cuda.is_available() else "cpu"
+    accelerator = "gpu" if cuda_available else "cpu"
     trainer_args = dict(default_root_dir=tmpdir, max_epochs=7, accelerator=accelerator, devices=1)
     model = RegressionModel()
     qmodel = copy.deepcopy(model)

--- a/tests/standalone_tests.sh
+++ b/tests/standalone_tests.sh
@@ -28,14 +28,14 @@ files=$(echo "$grep_output" | cut -f1 -d: | sort | uniq)
 # get the list of parametrizations. we need to call them separately. the last two lines are removed.
 # note: if there's a syntax error, this will fail with some garbled output
 if [[ "$OSTYPE" == "darwin"* ]]; then
-  parametrizations=$(pytest $files --collect-only --quiet "$@" | tail -r | sed -e '1,3d' | tail -r)
+  parametrizations=$(python -m pytest $files --collect-only --quiet "$@" | tail -r | sed -e '1,3d' | tail -r)
 else
-  parametrizations=$(pytest $files --collect-only --quiet "$@" | head -n -2)
+  parametrizations=$(python -m pytest $files --collect-only --quiet "$@" | head -n -2)
 fi
 parametrizations_arr=($parametrizations)
 
 # tests to skip - space separated
-blocklist='tests/profiler/test_profiler.py::test_pytorch_profiler_nested_emit_nvtx'
+blocklist='tests/profiler/test_profiler.py::test_pytorch_profiler_nested_emit_nvtx tests/utilities/test_warnings.py'
 report=''
 
 for i in "${!parametrizations_arr[@]}"; do

--- a/tests/utilities/test_warnings.py
+++ b/tests/utilities/test_warnings.py
@@ -23,7 +23,7 @@ from pytorch_lightning.utilities.rank_zero import _warn, rank_zero_deprecation, 
 from pytorch_lightning.utilities.warnings import WarningCache
 
 standalone = os.getenv("PL_RUN_STANDALONE_TESTS", "0") == "1"
-if standalone:
+if standalone and __name__ == "__main__":
 
     stderr = StringIO()
     # recording

